src/libDirectoryService/FinalBlockPreProcessing.cpp

```diff
@@ -489,9 +489,13 @@ bool DirectoryService::CheckLegitimacyOfMicroBlocks() {
 
   {
     lock_guard<mutex> g(m_mutexMicroBlocks);
+    std::set<MicroBlock> microBlocks;
+    if (m_microBlocks.find(m_mediator.m_currentEpochNum) !=
```

Comment:
Good call to add this check. However, the creation of `microBlocks` adds performance and memory penalty. Maybe you can instead encapsulate the for loop below within the check, so that no microblocks iteration is done if we can't find the epoch number in the map.

Reply:
This is not inside the loop

Reply:
Updated